riscv/zvk_ext_macros.h

```diff
@@ -0,0 +1,1251 @@
+// the vector number from which is was extracted.
+// The vector number is formatted with "%u", the EGU32x4_t value
+// with the  PRIxEGU32x4 documented above.
+#define PRI_uREG_xEGU32x4 "v%" PRIuREG "(" PRIxEGU32x4 ")"
+#define PRV_REG_EGU32x4_BE(reg_num, reg) (reg_num), PRVEGU32x4_BE(reg)
+#define PRV_REG_EGU32x4_LE(reg_num, reg) (reg_num), PRVEGU32x4_LE(reg)
+
+// Helper macros to print a EGU32x8_t element group, together with
+// the vector number from which is was extracted.
+// The vector number is formatted with "%u", the EGU32x8_t value
+// with the PRIxEGU32x8 documented above.
+#define PRI_uREG_xEGU32x8 "v%" PRIuREG "(" PRIxEGU32x8 ")"
+#define PRV_REG_EGU32x8_LE(reg_num, reg) (reg_num), PRVEGU32x8_LE(reg)
+
+//
+// Predicate Macros
+//
+
+// Ensures that the ZVBB extension (vector crypto bitmanip) is present,
+// which depends on vector.
+#define require_zvbb \
+  do { \
+    require_vector(true); \
+    require_extension(EXT_ZVBB); \
+  } while (0)
+
+// Ensures that the ZVBC extension (vector carryless multiplication)
+// is present, which depends on vector.
+#define require_zvbc \
+  do { \
+    require_vector(true); \
+    require_extension(EXT_ZVBC); \
+  } while (0)
+
+// Ensures that the ZVKG extension (vector Gallois Field Multiplication)
+// is present, which depends on vector.
+#define require_zvkg \
+  do { \
+    require_vector(true); \
+    require_extension(EXT_ZVKG); \
+  } while (0)
+
+// Ensures that a ZVK extension supporting SHA-256 is present.
+// For SHA-256, this support is present in either Zvknha or Zvknhb.
+// Also ensures that vector support is present.
+#define require_zvknh_256 \
+  do { \
+    require_vector(true); \
+    require_either_extension(EXT_ZVKNHA, EXT_ZVKNHB); \
+  } while (0)
+
+// Ensures that the ZVKNED extension (vector AES single round) is present,
+// which depends on vector.
+#define require_zvkned \
+  do { \
+    require_vector(true); \
+    require_extension(EXT_ZVKNED); \
+  } while (0)
+
+// Ensures that a ZVK extension supporting SHA-512 is present.
+// For SHA-512, this support is only present in Zvknhb.
+// Also ensures that vector support is present.
+#define require_zvknh_512 \
+  do { \
+    require_vector(true); \
+    require_extension(EXT_ZVKNHB); \
+  } while (0)
+
+// Ensures that the ZVKSED extension (vector SM4 block cipher)
+// is present, which depends on vector.
+#define require_zvksed \
+  do { \
+    require_vector(true); \
+    require_extension(EXT_ZVKSED); \
+  } while (0)
+
+// Ensures that the ZVKSH extension (vector SM3 hash) is present,
+// which depends on vector.
+#define require_zvksh \
+  do { \
+    require_vector(true); \
+    require_extension(EXT_ZVKSH); \
+  } while (0)
+
+// Ensures that the vector instruction is not using a mask.
+#define require_no_vmask  require(insn.v_vm() == 1)
+
+// Ensures that the current VSEW is the given N number of bits.
+#define require_vsew(N)  require(P.VU.vsew == (N))
```

Comment:
These two are so trivial and self-expanatory that I'd say they should just be inlined.

Reply:
I agree for `require_vsew()` and got rid of them.

I have a stronger attachment to `require_no_vmask` as I find it a lot clearer than `require(insn.v_vm() == 1)`, especially since VM is a highly overloaded acronym and "VM == 1" meaning "vector mask is not used" is very counter-intuitive.  
It is a case where someone intimate with the vector spec may find the "insn.v_vm() == 1" very clear but newcomers or more casual contributors will either misinterpret or need to go back to the spec to understand. When I wrote that code I very much needed the more explicit semantics.

Reply:
Let me know if you still want me to remove the `require_no_vmask`.